app/src/main/java/com/github/freeman/bootcamp/games/guessit/DisplayRecapsActivity.kt

```diff
@@ -0,0 +1,146 @@
+                    .fillMaxSize()) {
+                    Column(modifier = Modifier.background(color = MaterialTheme.colorScheme.background)) {
+                        TopAppbarDisplayRecaps()
+                        VideoGallery(videos = videos)
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun TopAppbarDisplayRecaps(context: Context = LocalContext.current) {
+
+    TopAppBar(
+        modifier = Modifier.testTag("topAppbarDisplayRecaps"),
+        title = {
+            Text(
+                text = "Recaps",
+                maxLines = 1,
+                overflow = TextOverflow.Ellipsis,
+                style = MaterialTheme.typography.titleMedium,
+                color = MaterialTheme.colorScheme.primary,
+                fontSize = 20.sp
+            )
+        },
+        backgroundColor = MaterialTheme.colorScheme.background,
+        elevation = 4.dp,
+        navigationIcon = {
+            IconButton(onClick = {
+                val activity = (context as? Activity)
+                activity?.finish()
+            }) {
+                Icon(
+                    Icons.Filled.ArrowBack,
+                    contentDescription = "Go back",
+                    tint = MaterialTheme.colorScheme.primary
+                )
+            }
+        }
+    )
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun VideoGallery(videos: Array<File>?) {
+    val pagerState = rememberPagerState()
+    val context = LocalContext.current
+
+    if (videos != null) {
+        Box(
+            modifier = Modifier
+                .padding(5.dp)
+                .background(color = MaterialTheme.colorScheme.background)
+                .testTag("videoGallery")
+        ) {
+            Column {
+                VerticalPager(
+                    modifier = Modifier
+                        .background(color = MaterialTheme.colorScheme.background),
+                    pageCount = videos.size,
+                    state = pagerState
+                ) { page ->
+                    if (page < videos.size) {
+                        val videoFile = videos[page]
+                        val videoUri = videoFile.absolutePath
+                        Box(
+                            modifier = Modifier
+                                .padding(16.dp)
+                                .fillMaxWidth()
+                                .background(
+                                    MaterialTheme.colorScheme.primaryContainer,
+                                    shape = RoundedCornerShape(16.dp)
+                                )
+                                .padding(16.dp)
+                        ) {
+                            Column {
+                                RecapPreview(
+                                    videoUrl = videoUri
+                                )
+                                MainMenuButton(
+                                    testTag = "shareRecap",
```

Comment:
Same remark as before for the testTag, maybe extract it in a variable in `strings.xml`.